src/probnum/quad/_integration_measures.py

```diff
@@ -0,0 +1,236 @@
+        self.random_state = random_state
+        self.name = name
+
+    def __call__(self, points: Union[float, np.floating, np.ndarray]) -> np.ndarray:
+        """Evaluate the density function of the integration measure.
+
+        Parameters
+        ----------
+        points :
+            (n_points,) or (n_points, dim)
+
+        Returns
+        -------
+        density_evals :
+            (n_points,) or (n_points,dim)
+        """
+        return self.random_variable.pdf(points).squeeze()
+
+    def sample(self, n_sample) -> np.ndarray:
+        """Sample ``n_sample`` points from the integration measure.
+
+        Parameters
+        ----------
+        n_sample :
+            Number of points to be sampled
+
+        Returns
+        -------
+        points :
+            (n_sample,) or (n_sample,dim)
+        """
+        return np.squeeze(self.random_variable.sample(size=n_sample))
+
+    def _set_dimension_domain(self, dim, domain):
+        """Sets the integration domain and dimension.
+
+        The following logic is used to set the domain and dimension:
+            1. If ``dim`` is not given (``dim == None``):
+                1a. If either ``domain[0]`` or ``domain[1]`` is a scalar, the dimension
+                    is set as the maximum of their lengths and the scalar is expanded to
+                    a constant vector.
+                1b. Otherwise, if the ``domain[0]`` and ``domain[1]`` are not of equal
+                    length, an error is raised.
+            2. If ``dim`` is given:
+                2a. If both ``domain[0]`` and ``domain[1]`` are scalars, they are
+                    expanded to constant vectors of length ``dim``.
+                2b. If only one of `domain[0]`` and ``domain[1]`` is a scalar and the
+                    length of the other equals ``dim``, the scalar one is expanded to a
+                    constant vector of length ``dim``.
+                2c. Otherwise, if neither of ``domain[0]`` and ``domain[1]`` is a
+                    scalar, error is raised if either of them has length which does not
+                    equal ``dim``.
+        """
+        domain_a_dim = np.size(domain[0])
+        domain_b_dim = np.size(domain[1])
+
+        # Check that given dimensions match and are positive
+        dim_mismatch = False
+        if dim is None:
+            if domain_a_dim == domain_b_dim:
+                dim = domain_a_dim
+            elif domain_a_dim == 1 or domain_b_dim == 1:
+                dim = np.max([domain_a_dim, domain_b_dim])
+            else:
+                dim_mismatch = True
+        else:
+            if (domain_a_dim > 1 or domain_b_dim > 1) and dim != np.max(
+                [domain_a_dim, domain_b_dim]
+            ):
+                dim_mismatch = True
+
+        if dim_mismatch:
+            raise ValueError(
+                "Domain limits must have the same length or at least "
+                "one of them has to be one-dimensional"
```

Comment:
Super petty point: current style is that all error messages have a full stop.

Reply:
done